packages/contracts-bedrock/src/L1/SystemConfigInterop.sol

```diff
@@ -0,0 +1,86 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { OwnableUpgradeable } from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import { ISemver } from "src/universal/ISemver.sol";
+import { ResourceMetering } from "src/L1/ResourceMetering.sol";
+import { Storage } from "src/libraries/Storage.sol";
+import { Constants } from "src/libraries/Constants.sol";
+import { OptimismPortalInterop as OptimismPortal } from "src/L1/OptimismPortalInterop.sol";
+import { GasPayingToken, IGasToken } from "src/libraries/GasPayingToken.sol";
+import { ERC20 } from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import { ConfigType } from "src/L2/L1BlockInterop.sol";
+import { SystemConfig } from "src/L1/SystemConfig.sol";
+
+/// @title SystemConfigInterop
+/// @notice The SystemConfig contract is used to manage configuration of an Optimism network.
+///         All configuration is stored on L1 and picked up by L2 as part of the derviation of
+///         the L2 chain.
+contract SystemConfigInterop is SystemConfig {
+    /// @notice Error thrown when the OptimismPortal address is not set.
+    error OptimismPortalNotSet();
+
+    constructor() SystemConfig() { }
+
+    /// @custom:semver 2.2.0+interop
+    function version() public pure override returns (string memory) {
+        return string.concat(super.version(), "+interop");
+    }
+
+    /// @notice Internal setter for the gas paying token address, includes validation.
+    ///         The token must not already be set and must be non zero and not the ether address
+    ///         to set the token address. This prevents the token address from being changed
+    ///         and makes it explicitly opt-in to use custom gas token. Additionally,
+    ///         OptimismPortal's address must be non zero, since otherwise the call to set the
+    ///         config for the gas paying token to OptimismPortal will fail.
+    /// @param _token Address of the gas paying token.
+    function _setGasPayingToken(address _token) internal override {
+        if (_token != address(0) && _token != Constants.ETHER && !isCustomGasToken()) {
+            if (optimismPortal() == address(0)) revert OptimismPortalNotSet();
```

Comment:
Were you hitting this issue in practice?

Reply:
not in practice, but it's possible that `_setGasPayingToken` is called for a custom token and, if `optimismPortal` is set to the zero address, the call to `optimismPortal` reverts with the error `0 bytes of code`. the error would make it more explicit that the call is reverting due `optimismPortal` not being set. what do you think?

Reply:
To get the effect that you are looking for, it would actually be more accurate to do a codesize check rather than a "if equals zero value" check because you can have it be `address(0x01)` for example and then have it fail still with no error message

Reply:
I think its fine to have this check altho I could imagine there being more calls to other contracts in the future so I recommend making it `error NoCode(address)` or error NoCode(string name)`

Reply:
good idea!